packages/waku/src/lib/utils/merge-vite-config.ts

```diff
@@ -0,0 +1,24 @@
+import type { UserConfig } from 'vite';
+import {
+  resolveConfig as resolveViteConfig,
+  mergeConfig as mergeViteConfig,
+} from 'vite';
+
+export async function mergeUserViteConfig(config: UserConfig) {
```

Comment:
It would be nice to add a comment about this workaround is required only (?) for hmr. (Can be a separate PR.)